tests/integration/setup_environment.py

```diff
@@ -0,0 +1,86 @@
+from testcontainers.core.waiting_utils import wait_container_is_ready, wait_for_logs
+from testcontainers.postgres import PostgresContainer
+from testcontainers.core.container import DockerContainer
+
+
+postgres_image_name = "postgres:latest"
+adcm_image_name = "hub.adsw.io/adcm/adcm:develop"
+adcm_port = 8000
+postgres_port = 5432
+container_name = "test_adcm"
+postgres_name = "test_pg_db"
+db_user = "adcm"
+db_name = "adcm"
+db_password = "password"
+
+
+
+class ADCMPostgresContainer(PostgresContainer):
+    def __init__(self, image: str):
+        super().__init__(image)
+
+    def setup_postgres(self, username: str, password: str, adcm_db_name: str, container_network: Network) -> None:
+        self.with_name(postgres_name)
+        self.with_bind_ports(postgres_port, postgres_port)
+        self.password = password
+        self.with_network(container_network)
+
+        self.start()
+
+        init_db_cmds = f"CREATE USER {username} WITH ENCRYPTED PASSWORD '{password}'; CREATE DATABASE {adcm_db_name} OWNER {username};"
+        self.with_command(
+            f"PGPASSWORD={self.password} psql --username {self.username} --host 127.0.0.1 -c '{init_db_cmds}'")
+
+
+class ADCMContainer(DockerContainer):
+    def __init__(self, image: str):
+        super().__init__(image)
+
+    def setup_container(self, container_network: Network):
+        self.with_network(container_network)
+        self.with_bind_ports(adcm_port, adcm_port)
+        self.start()
+
+
+class SetupManager:
+    def __init__(self, postgres_image: str, adcm_image: str):
+        self.network = Network()
+        self.postgres_container = ADCMPostgresContainer(postgres_image)
+        self.adcm_container = ADCMContainer(adcm_image)
+
+    def __enter__(self):
+        # Start the network
+        self.network.__enter__()
+
+        # Setup and start the postgres container
+        self.postgres_container.with_name(postgres_name)
+        self.postgres_container.with_bind_ports(postgres_port, postgres_port)
+        self.postgres_container.with_network(self.network)
+        self.postgres_container.start()
+
+        # Initialize the PostgreSQL database
+        init_db_cmds = (
+            f"CREATE USER {db_user} WITH ENCRYPTED PASSWORD '{db_password}'; "
+            f"CREATE DATABASE {db_name} OWNER {db_user};"
+        )
+        self.postgres_container.with_command(
+            f"PGPASSWORD={self.postgres_container.password} psql --username {self.postgres_container.username} --host 127.0.0.1 -c '{init_db_cmds}'"
+        )
+
+        # Setup and start the Docker container
+        self.adcm_container.with_network(self.network)
+        self.adcm_container.with_bind_ports(adcm_port, adcm_port)
+        self.adcm_container.start()
+
+        wait_container_is_ready(self.adcm_container)
+        wait_container_is_ready(self.postgres_container)
+        wait_for_logs(self.postgres_container, "database system is ready to accept connections")
```

Comment:
feels lukewarm this line should ho before adcm start or there's chance migrations will fail

Reply:
changed